src/AggregateRoots/AggregateRoot.php

```diff
@@ -116,6 +125,25 @@ public function recordThat(ShouldBeStored $domainEvent): static
         return $this;
     }
 
+    public function recordConcurrently(ShouldBeStored $domainEvent, bool|callable $allowConcurrent = true): static
+    {
```

Comment:
One idea I had for a potential API to make it a bit more easy, is to not have a separate `recordConcurrently` method, but instead make the `record` method able to accept a special object `ConcurrentEvent`. You could add a helper method to convert a `ShouldBeStored` to such a special `ConcurrentEvent` object, it could simply be a wrapper as well, if you want to keep things simpler.

```php
$ar->recordThat(concurrent(new MoneyAdded(…)))
```

Just an idea :)

Reply:
Think I'm going to hold off and keep the API separate for now. In the next major version I want a single API for all as described above. Storing a different type of object internally also isn't easy without changing the contents of `$recordedEvents`, which could be considered a breaking change.